src/main/java/org/jfrog/gradle/plugin/artifactory/utils/PublicationUtils.java

```diff
@@ -255,6 +260,42 @@ public static void extractMavenDeployDetails(MavenPublicationInternal mavenPubli
         extractMavenArtifacts(destination, publicationName, mavenPublicationInternal, mavenNormalizedPublication);
     }
 
+    /**
+     * Extract archive configuration artifacts, creates deploy details for it and stores them at the given destination
+     *
+     * @param configuration - configuration containing the artifacts to publish
+     * @param publisher     - publisher handler of the project
+     * @param destination   - task to collect and store the created details
+     */
+    public static void extractArchivesDeployDetails(Configuration configuration, ArtifactoryClientConfiguration.PublisherHandler publisher, ArtifactoryTask destination) {
+        Project project = destination.getProject();
+        PublishArtifactSet artifacts = configuration.getAllArtifacts();
+        for (PublishArtifact artifact : artifacts) {
+            File file = artifact.getFile();
+            DeployDetails.Builder builder = createArtifactBuilder(file, configuration.getName());
+            if (builder == null) {
+                continue;
+            }
+            String gid = project.getGroup().toString();
+            if (publisher.isM2Compatible()) {
+                gid = gid.replace(".", "/");
+            }
+            Map<String, String> extraTokens = new HashMap<>();
+            if (StringUtils.isNotBlank(artifact.getClassifier())) {
+                extraTokens.put("classifier", artifact.getClassifier());
+            }
+            String artifactPath = IvyPatternHelper.substitute(publisher.getIvyArtifactPattern(), gid, project.getName(),
```

Comment:
IvyArtifactPattern can be optional right? Should we check for that before?

Reply:
getIvyArtifactPattern` returns a default non-blank value if the IvyArtifactPattern is blank:
```java
public String getIvyArtifactPattern() {
    String value = getStringValue(IVY_ART_PATTERN);
    if (StringUtils.isBlank(value)) {
        return LayoutPatterns.M2_PATTERN;
    }
    return value.trim();
}
```